### PR TITLE
Unexpected keyword argument

### LINKED ISSUE #48855: STYLE fix pylint issues
In https://github.com/pandas-dev/pandas/pull/48759/files we're introduced `pylint`, but have turned off its warnings as there's a _lot_ of them:

https://github.com/pandas-dev/pandas/blob/d719840e5a2639babab3b4646b11a111547c518f/pyproject.toml#L36-L67

Task here is:

1. pick one of the above warnings / errors, and remove it from `pyproject.toml`
2. run `pre-commit run pylint --hook-stage manual --all-files`. If this already passes, skip to step 4.
3. fixup any warnings that result
4. if `pre-commit run pylint --hook-stage manual --all-files` passes, then stage and commit your changes and open a pull request

If you believe a warning is a false-positive, then it's OK to ignore it in-line, e.g. 
```python
df.rename(id, mapper=id)  # pylint: disable=redundant-keyword-arg
```

Please comment here which pylint warning you'll work on before starting so we don't duplicate work. No need to ask for permission to work on this, and no need to comment "take" as multiple people can work on this concurrently

---

NOTE: please do _not_ write "closes #48855" in your PR

---

checks we probably can't turn off:
- too-many-function-args
- unsubscriptable-object
- unsupported-assignment-operation
- no-name-in-module
- no-member
- import-error
- not-an-iterable
- invalid-unary-operator-type
- invalid-name
- wrong-import-order (and anything else related with imports)
- unexpected-keyword-argument
- use-implicit-booleaness-not-len
- use-implicit-booleaness-not-comparison
- comparison-with-itself
- no-else-*
- line-too-long
- overridden-final-method
- singleton-comparison
- unsupported-membership-test
- pointless-statement
- broad-except
- undefined-variable

If you're not sure whether a check should fit into the above list, please do ask

### PR DESCRIPTION
**Pull request by shalearkane**
This fixed unexpected-keyword-argument warnings from pylint on one file.
The other warnings produced were already expected to be caught in tests.

- [x] xref #48855 one of many style checks, still needs work.

### CODE REVIEW

**Comment by shalearkane** [comment]
`pre-commit` is failing because I am using Optional type annotation with the older syntax `Optional[type]` as the newer syntax `type | None` is only supported on Python 3.10+

**Comment by MarcoGorelli** [comment]
> `pre-commit` is failing because I am using [Optional type annotation](https://docs.python.org/3/library/typing.html#typing.Optional) with the older syntax `Optional[type]` as the newer syntax `type | None` is only supported on Python 3.10+

it's supported with `from __future__ import annotations`, which is at the top of the file

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -738,7 +738,12 @@ def _with_infer(cls, *args, **kwargs):
         return result
 
     @cache_readonly
-    def _constructor(self: _IndexT) -> type[_IndexT]:
+    def _constructor(
+        self: _IndexT,
+        name: Hashable | None = None,
+        copy: bool | None = None,
+        dtype=None,
```
[the remark this answers is not in the document]
Comment (shalearkane) [reply pandas/core/indexes/base.py:745]: Yes, I looked into the code, it seems like _constructor is being overridden by the classes `ExtensionIndex`, `MultiIndex` and `NumericIndex` which inherit form the base class `Index`.

[the remark this answers is not in the document]
Comment (shalearkane) [reply pandas/core/indexes/base.py:745]: classes

[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply pandas/core/indexes/base.py:745]: wait, this isn't resolved - I think the issue is that pylint can't tell that `@cache_readonly` makes `_constructor` act like a property and that it doesn't take arguments

I'll have another look later

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @shalearkane for your PR

I've had a closer look, and I don't think we should enable this check

`cache_readonly` acts like a property, but `pylint` isn't aware of this and so it thinks that the arguments that are passed to whetever function the property returns are being passed to the property itself

`mypy` already checks this, and the way pandas gets mypy to recognise that `cache_readonly` is a property is

https://github.com/pandas-dev/pandas/blob/7440fe27eef0aab0f217ca9fb434f1e3ac74836e/pandas/_libs/properties.pyi#L16

So, I'll add this to the list of checks we probably can't turn off

Closing for now then

**Pull request closed without merging**